x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authz/AuthorizationService.java

```diff
@@ -498,39 +511,63 @@ private void authorizeAction(
                     return SubscribableListener.newSucceeded(resolvedIndices);
                 } else {
                     final SubscribableListener<ResolvedIndices> resolvedIndicesListener = new SubscribableListener<>();
+                    final var authorizedIndicesListener = new SubscribableListener<AuthorizationEngine.AuthorizedIndices>();
+                    authorizedIndicesListener.<Tuple<AuthorizationEngine.AuthorizedIndices, TargetProjects>>andThen(
+                        (l, authorizedIndices) -> {
+                            if (request instanceof IndicesRequest.Replaceable replaceable
+                                && ResponseValidator.shouldResolveCrossProject(replaceable)) {
```

Comment:
Nit: can we let `shouldResolveCrossProject` handle the `instanceof` check internally?

Reply:
I wanted to do that originally but `shouldResolve(Replaceable)` and `shouldResolve(TransportRequest)` results in ambiguous calls for requests that implement both. I'll just add a `shouldResolve...TransportRequest()` to disambiguate